### DISCUSSION TITLE
Static methods not being found in VSCode

### DISCUSSION THREAD

**Discussion by DanjBethel** [spatie/laravel-query-builder#732] [Q&A] [answered]
Getting "Non static method 'for' should not be called statically."

Laravel: 9
PHP: 8.1
Laravel-query-builder: 5.0

Not sure what I could be doing wrong. I just installed the package and tried using it and VsCode started barking at me. Also receiving a 500 error in my test saying the same.

**Comment by DanjBethel** [comment, marked as answer]
Any help here would be greatly appreciated. Thank you. @freekmurze

**Reply by dominikb** [reply]
Can you show how you are invoking the `QueryBuilder`?

Edit: I just tried setting up a new Laravel project and querying users worked just fine:
```
composer create-project laravel/laravel test-laravel-query-builder
cd test-laravel-query-builder
composer require spatie/laravel-query-builder

// Set up a db, e.g. sqlite and change parameters in .env
touch database/database.sqlite
echo "DB_CONNECTION=sqlite" >> .env
echo "DB_DATABASE=\"`realpath database/database.sqlite`\"" >> .env
php artisan migrate 

php artisan serve
```

Change `routes/web.php`:
```php
Route::get('/', function () {
    return Spatie\QueryBuilder\QueryBuilder::for(App\Models\User::class)
        ->allowedFields("name")
        ->get();
});
```

Then add a user to your database and you can query `http://localhost:8000/?fields[users]=name` just fine.

Laravel: 9.1.0
laravel-query-builder: 5.0
PHP: 8.1.2

**Reply by DanjBethel** [reply]
Laravel: 9.0.1 (to be specific)

```
use Spatie\QueryBuilder\QueryBuilder;

$users = QueryBuilder::for(User::class)
            ->allowedFilters(['first_name', 'last_name', 'email'])
            ->orderBy('last_name')
            ->paginate();
```

I originally thought it was a VSCode error but like I said, running tests also give me the same error as a 500 code. Also I am trying to implement into an already working application so, database has already been properly configured.

**Reply by dominikb** [reply]
Can you find out what exception causes this response? A 500 status code is a default response if an unhandled exception is raised somewhere in the request-response cycle.

**Reply by DanjBethel** [reply]
```
• Tests\Feature\UsersTest > it can retrieve users by query
  Expected response status code [200] but received 500.

  The following exception occurred during the request:

  Error: Non-static method Spatie\QueryBuilder\QueryBuilder::for() cannot be called statically in /var/www/html/app/Http/Controllers/UserController.php:23
```

Followed by a long stack trace. Would seeing this help too?

Here is the entire index method in UserController

```
    public function index()
    {
        // $search = Str::lower(request('search'));

        // $users = User::search($search)
        // ->orderBy('last_name')
        // ->paginate();

        $users = QueryBuilder::for(User::class)
                ->allowedFilters(['first_name', 'last_name', 'email'])
                ->orderBy('last_name')
                ->paginate();
        

        return UserResource::collection($users);
    }
```

**Reply by dominikb** [reply]
This is quite odd as the class clearly contains the static method you are trying to call: 
https://github.com/spatie/laravel-query-builder/blob/080d923aabef57a13e6c40a229bc55b6204e41a2/src/QueryBuilder.php#L94

Have you, by any chance, edited the `QueryBuilder.php` file? You could try deleting the `vendor` directory and calling `composer install` again.

**Reply by DanjBethel** [reply]
Ahhh I see what happened. Honest mistake.

So, I have a php formatter for VsCode. phpfmt.

I opened the QueryBuilder class previously just to take a look around, as a force of habit, I hit CMD+S. That for some reason removed the static keyword. Your suggestion of deleting the vendor directory and calling composer install again worked.

Thank you so much.
